operator/main.go

```diff
@@ -148,6 +159,78 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	operatorID, err := os.Hostname()
+	if err != nil {
+		log.WithError(err).Fatal("failed to get hostname")
+	}
+	operatorID = rand.RandomStringWithPrefix(operatorID+"-", 10)
+
+	ns := option.Config.K8sNamespace
+	// If due to any reason the NAMESPACE is not set we assume it to be
+	// in default namespace.
+	if ns == "" {
+		ns = metav1.NamespaceDefault
+	}
+
+	// Use a Go context so we can tell the leaderelection code when we
+	// want to step down
+	ctx, contextCancelFunc := context.WithCancel(context.Background())
```

Comment:
Will this contextCancelFunc variable be shadowed (i.e. it's new local variable instead of package level variable) ?

Reply:
I just quickly try in playground https://play.golang.org/p/qSPp12Ps0XW

Reply:
Yeah, you are right :sweat_smile: . I will push the required changes.